File: programmers/kakao/2022/신고 결과 받기.py
def solution(id_list, report, k):
    
    id_dict = dict()
    report_dict = dict()
    report_check = []
    
    for id in id_list:
        id_dict[id] = []
        report_dict[id] = 0
    
    for val in report:
        ids = val.split(" ")
        report_check.append((ids[0], ids[1]))
        
    report_check = list(set(report_check))
    
    for id in report_check:
        id_dict[id[0]].append(id[1])
        report_dict[id[1]] += 1
    
    report_dict = dict(filter(lambda x: x[1]>=k, report_dict.items())).keys()
    
    answer = [0] * len(id_dict)
    
    for report_id in report_dict:
        for idx, ids in enumerate(id_dict.values()):
            if report_id in ids:
                answer[idx] += 1
    
    return answer

File: programmers/kakao/2022/신고 결과 받기.swift
import Foundation

func solution(_ id_list:[String], _ report:[String], _ k:Int) -> [Int] {
    var reportedIDDict: [String: ([String], Int)] = [:]
    var IDIdxDict: [String: Int] = [:]
    var result: [Int] = Array(repeating: 0, count: id_list.count)

    for (idx, id) in id_list.enumerated() { 
        reportedIDDict.updateValue(([], 0), forKey: id) 
        IDIdxDict.updateValue(idx, forKey: id)
    }

    report.forEach {
        let reporterUser = $0.split(separator: " ").map { String($0) }
        let reporter = reporterUser[0], user = reporterUser[1]

        if !reportedIDDict[user]!.0.contains(reporter) { 
            var tempArr = reportedIDDict[user]!.0 + [reporter]
            var tempCnt = reportedIDDict[user]!.1 + 1

            reportedIDDict.updateValue((tempArr, tempCnt), forKey: user)  
        }
    }

    for (reporterArr, cnt) in Array(reportedIDDict.values) {
        if cnt >= k {
            reporterArr.forEach { result[IDIdxDict[$0]!] += 1 }
        }
    }

    return result
}
